src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,21 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User cannot be null");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new RegistrationException("Login must be at least 6 characters");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new RegistrationException("Password must be at least 6 characters");
+        }
+        if (user.getAge() == null || user.getAge() < 18) {
+            throw new RegistrationException("User must be at least 18 years old");
```

Comment:
The age validation should also be split into separate conditions. One for checking if the age is null and another one for checking if the age is below the minimum. This will help to provide more informative exception messages.